### PR TITLE
New test vscsi dlpar

### PR DESCRIPTION
**Pull request by smruti77**
new test for dlpar operation of vscsi device.

Signed-off-by: bismurti bidhibrata pattajoshi <bbidhibr@in.ibm.com>

### CODE REVIEW

**Comment by smruti77** [comment]
[vscsi_dlpar.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/4454986/vscsi_dlpar.log)

File: io/disk/dlpar_vscsi.py [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+            self.log.info(output)
+        except CommandFailed as cf:
+            self.fail("failed dlpar operation:remove disk, %s" % cf)
+        cmd = "mkvdev -vdev %s -vadapter %s -dev %s" % (self.hdisk_name,
+                                                        self.host, self.vscsi)
+        try:
+            output = self.run_command(cmd)
+            self.log.info(output)
+        except CommandFailed as cf:
+            self.fail("Failed dllpar operation:add disk, %s" % cf)
```
Comment (narasimhan-v) [inline io/disk/dlpar_vscsi.py:151]: Better to have functions to remove and add device, and call them.
Also, can have number of dlpar operations controlled by yaml parameter

File: io/disk/dlpar_vscsi.py [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+        for line in output.decode("utf-8").splitlines():
+            if self.disk in line:
+                value = line.split()[0]
+        for line in output.decode("utf-8").splitlines():
+            if value in line:
+                if "dir" in line:
+                    self.disk_dir = line.split()[-1]
+        cmd = r"cat %s/inquiry" % self.disk_dir
+        output = process.system_output(cmd, shell=True)
+        self.hdisk_name = output.split()[2].strip(b'0001').decode("utf-8")
```
Comment (narasimhan-v) [inline io/disk/dlpar_vscsi.py:85]: It can be decoded to utf-8 and then 0001 can be stripped

File: io/disk/dlpar_vscsi.py [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+        cmd = "lsscsi -vl | tr -d '[]'"
+        output = process.system_output(cmd, shell=True)
+        for line in output.decode("utf-8").splitlines():
+            if self.disk in line:
+                value = line.split()[0]
+        for line in output.decode("utf-8").splitlines():
+            if value in line:
+                if "dir" in line:
+                    self.disk_dir = line.split()[-1]
+        cmd = r"cat %s/inquiry" % self.disk_dir
```
Comment (narasimhan-v) [inline io/disk/dlpar_vscsi.py:83]: can't we use `genio.read_file()` ?

File: io/disk/dlpar_vscsi.py [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+            if self.disk in line:
+                self.slot = line.split()[-1].split('-')[-2]
+        cmd = "lsdev -slots"
+        output = self.run_command(cmd)
+        print(output)
+        for line in output.splitlines():
+            if self.slot in line:
+                self.host = line.split()[-1]
+        self.log.info(self.host)
+        cmd = "lsscsi -vl | tr -d '[]'"
```
Comment (narasimhan-v) [inline io/disk/dlpar_vscsi.py:74]: please use native python parsing instead of bash.

**Comment by smruti77** [comment]
> @smruti77 What are the differences between vFC and vscsi dlpar ?
> Based on that, we can decide if we need separate tests.

this is for scsi disk.

File: io/disk/dlpar_vscsi.py (hunk 1 above)
Comment (smruti77) [reply io/disk/dlpar_vscsi.py:151]: ok

File: io/disk/dlpar_vscsi.py (hunk 2 above)
Comment (smruti77) [reply io/disk/dlpar_vscsi.py:85]: if i will put decode("utf-8") "output = process.system_output(cmd, shell=True)" here getting error, so put after that.

File: io/disk/dlpar_vscsi.py (hunk 3 above)
Comment (smruti77) [reply io/disk/dlpar_vscsi.py:83]: ok,will use that.

File: io/disk/dlpar_vscsi.py (hunk 4 above)
Comment (smruti77) [reply io/disk/dlpar_vscsi.py:74]: ok

File: io/disk/dlpar_vscsi.py (hunk 2 above)
Comment (narasimhan-v) [reply io/disk/dlpar_vscsi.py:84]: oh ok

**Comment by narasimhan-v** [comment]
> > @smruti77 What are the differences between vFC and vscsi dlpar ?
> > Based on that, we can decide if we need separate tests.
> 
> this is for scsi disk.

Is there difference in how they are handled in this context ?

**Comment by smruti77** [comment]
> > > @smruti77 What are the differences between vFC and vscsi dlpar ?
> > > Based on that, we can decide if we need separate tests.
> > 
> > 
> > this is for scsi disk.
> 
> Is there difference in how they are handled in this context ?

i don't know more about vfc.we also thought to kept vfc, vscsi, veth in one file.letter we decided to have separate file, vfc was already their and i sent vscsi so.

File: io/disk/dlpar_vscsi.py [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+        cmd = r"cat %s/inquiry" % self.disk_dir
+        output = process.system_output(cmd, shell=True)
+        self.hdisk_name = output.split()[2].strip(b'0001').decode("utf-8")
+        self.log.info(self.hdisk_name)
+        cmd = "lsmap -all|grep -p %s" % self.hdisk_name
+        output = self.run_command(cmd)
+        for line in output.splitlines():
+            if "VTD" in line:
+                self.vscsi = line.split()[-1]
+        self.log.info(self.vscsi)
```
Comment (narasimhan-v) [inline io/disk/dlpar_vscsi.py:91]: @smruti77 Can we cancel the test in case self.vscsi is not got for some reason ?

Comment (smruti77) [reply io/disk/dlpar_vscsi.py:91]: ok

**Pull request merged**